### DISCUSSION TITLE
Release 2.0 - final call for feedback

### DISCUSSION THREAD

**Discussion by axxel** [zxing-cpp/zxing-cpp#470] [Announcements]
EDIT: 2.0 has been released. In case you find an issue with it, please report it as a new github issue. It will then have to be addressed in the next release.

-------
I'm (finally) about to make the 2.0 release. I already bumped the cmake project version but have not tagged it, yet.

I'd appreciate any feedback, especially last minute bug reports / regressions (my latest cmake/pkg-config cleanup might be subject to those).

Build artifacts can be found here: [winrt](https://github.com/zxing-cpp/zxing-cpp/actions/runs/3825730043#artifacts), [python](https://github.com/zxing-cpp/zxing-cpp/actions/runs/3825728129#artifacts), [android, ios, wasm](https://github.com/zxing-cpp/zxing-cpp/actions/runs/3825724325#artifacts)

Not sure who of you would see this new "Discussion" entry, hence the shout-out: @huycn @gitlost @vkrause @markusfisch @parallaxe @duanqn. In case you tested the code and have nothing to complain about, please leave a note or a thumbs up. If you are interested in giving it a go but you'll need more time than 3 days, say, also let me know, so I'll wait. Thanks.

**Comment by duanqn** [comment]
Tested and didn't see obvious issues. I also found the image with longest execution time (2s on my machine; commit 4201ed63b5e6766121f062d8afa221c6e0b51b0e; leaving all parameters as default except overriding downscale_factor to 2). Maybe it can be useful for you to spot some perf issues.

[slowest.zip](https://github.com/zxing-cpp/zxing-cpp/files/10334905/slowest.zip)

**Reply by axxel** [reply]
I checked your slowest.jpg and did not find anything particularly outstanding regarding the performance, it is simply a pretty big image (13453x8322 pixels). If you knew beforehand that you are only interested in UPC-A (or all LinearCodes for that matter) you could reduce the runtime to about 1/4.

**Reply by duanqn** [reply]
My bad I didn't check the dimensions. I was only looking at the size of the file.

**Comment by duanqn** [comment]
BTW I can't see https://github.com/zxing-cpp/zxing-cpp/releases/tag/untagged-957557d27e324b576ec0 . It gives me a 404 error.

**Reply by axxel** [reply]
thanks, see edited initial post above.

**Comment by markusfisch** [comment]
Can't see any serious problems, too 👍 

PS, I'am also getting a 404 for https://github.com/zxing-cpp/zxing-cpp/releases/tag/untagged-957557d27e324b576ec0

**Comment by vkrause** [comment]
Tested on Linux, I get 3 deviations in our test suite (additionally recognized things, ie. good), no noticeable performance difference. Android builds fine as well, I haven't done runtime testing there yet, but I'd not expect that to be different from Linux.

One minor change needed in the build system due to a too strict version check not accepting the new major version, but that's our fault.

**Reply by axxel** [reply]
Awesome. Thx.

**Reply by axxel** [reply]
Those 3 additionally recognized symbols, can you tell me which symbology they are from?

**Reply by vkrause** [reply]
Those are Aztec codes. Pure black/white input, but apparently slightly distorted with the individual elements varying between 4 and 5 pixel in height/width.

**Reply by axxel** [reply]
That's what I would have expected, since I worked on the Aztec detector code. Out of curiosity:
* do you have more Aztec samples that don't work yet?
* are those new working samples decoded with the `isPure` flag set?

**Reply by vkrause** [reply]
I don't think we have non-working Aztec samples anymore, the pure but distorted case was the last one, after the non-pure off-center ones. And yes, this was using the `isPure` flag.

The main remaining issue we have now are a few QR and Code 128 codes with no or insufficient quiet zones. That's a side-effect from extracting those from PDF documents (where the quiet zone is implied by being rendered on a white background with sufficient margins). But that's something we need to solve on our side, that's not ZXing's fault.

**Reply by axxel** [reply]
That bit about the missing quiet zones from extracted symbols surprises me because the library is (supposed to) treating 'no pixel' as a 'white pixel', i.e. should be able to crop the image to only contain the symbol without any quiet zone and it is still supposed to work. If you care to discuss this matter further and find a sample you can publish, please open a new issue.

**Reply by vkrause** [reply]
Good to know! I do also have samples that work despite an insufficient quite zone, so that certainly works. The ones that don't work can be made to work by manually adding a sufficient quiet zone, which is why I suspect it's related to that, but it could of course also be hitting another issue just by chance. I'll see if I have samples that can be published.

**Reply by vkrause** [reply]
Added issue https://github.com/zxing-cpp/zxing-cpp/issues/480 for the Code 128 case. Looking for a sample for the QR case uncovered that this was actually a bug in our PDF vector barcode processing code, ZXing has no problem decoding those in pure mode with non-compliant quiet zones after all.

**Comment by zander** [comment]
Tested this on Android (statically linked). I only process QR codes, from camera via Qt6-multimedia, though. Worked fine. :+1:

**Comment by ChristianNorbertBraun** [comment]
Can't see any problems with the iOS artefacts.
